bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -855,7 +915,8 @@ func (r *KubeadmConfigReconciler) storeBootstrapData(ctx context.Context, scope
 			},
 		},
 		Data: map[string][]byte{
-			"value": data,
+			"value":  data,
+			"format": []byte(scope.Config.Spec.Format),
```

Comment:
I'm still not sure about having the format field in here. Why is this needed?

Reply:
So user do not have to specify the bootstrap format explicitly for both infrastructure provider and bootstrap provider. This is needed when infrastructure provider requires extra handling of Ignition (as in case of AWS).

Reply:
>This is needed when infrastructure provider requires extra handling of Ignition (as in case of AWS).

Could you elaborate the extra handling needed in AWS? Thanks!

Reply:
>Could you elaborate the extra handling needed in AWS? Thanks!

Please see https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2271 for more details about that.

Reply:
Thanks! I added some thoughts https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2271#issuecomment-819575200

Reply:
@vincepri can we mark it as resolved?